core/src/input/input.stories.tsx

```diff
@@ -44,15 +64,37 @@ interface Props {
 	readOnly?: boolean;
 }
 
-export const Primary = (props: Props) => (
-	<Input
-		type={props.type}
-		style={Input.styles[props.style]}
-		size={Input.sizes[props.size]}
-		maxLength={props.maxLength}
-		disabled={props.disabled}
-		readOnly={props.readOnly}
-	/>
-);
+export const Primary = (props: Props) => {
+	const [text, setText] = useState("");
+	const onSubmit = (e: { preventDefault: () => void }) => {
+		e.preventDefault();
+		Dialog.alert(`You entered: ${text}`);
+	};
+	return (
+		<>
+			<form onSubmit={onSubmit}>
+				<div>
+					<Input
+						type={props.type}
+						style={Input.styles[props.style]}
+						size={Input.sizes[props.size]}
+						maxLength={props.maxLength}
+						disabled={props.disabled}
+						readOnly={props.readOnly}
+						setValue={(val: string) => setText(val)}
```

Comment:
You can pass the function itself to the prop:

```
setValue={setText}
```

Which is the same

Reply:
That being said, when setValue is set, you should also set "value={text}". It's a controlled component, completely. We should not support partially controlled component (one that has setValue but not value or vice versa)